src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+    useState<IAssetWithBalance | null>(values?.selectedAsset ?? null);
+
+  const [selectedAccountType, setSelectedAccountType] = useState<string>(
+    AccountTypes.Contract
+  );
+
+  const [selectedFromNetwork, setSelectedFromNetwork] = useState<Chain | null>(
+    null
+  );
+  const [keyBasedTotal, setKeyBasedTotal] = useState<number>(0);
+  const [smartWalletTotal, setSmartWalletTotal] = useState<number>(0);
+
+  const [totalKeyBasedPLRTokens, setTotalKeyBasedPLRTokens] = useState<number>(0);
+  const [totalSmartWalletPLRTokens, setTotalSmartWalletPLRTokens] = useState<number>(0);
+
+  const [accounts, setAccounts] = useState<{}[]>([]);
+  const [isNFTMember, setIsNFTMember] = useState<boolean>(false);
+
+  const hasEnoughPLR = totalKeyBasedPLRTokens >= MAX_PLR_TOKEN_LIMIT || totalSmartWalletPLRTokens >= MAX_PLR_TOKEN_LIMIT;
+
+  const fixed = multiCallData?.fixed ?? false;
+  const defaultCustomReceiverAddress =
+    values?.receiverAddress &&
+    !addressesEqual(providerAddress, values?.receiverAddress) &&
+    !addressesEqual(accountAddress, values?.receiverAddress)
+      ? values.receiverAddress
+      : null;
+  const [customReceiverAddress, setCustomReceiverAddress] = useState<
+    string | null
+  >(defaultCustomReceiverAddress);
+  const [useCustomAddress, setUseCustomAddress] = useState<boolean>(
+    !!defaultCustomReceiverAddress
+  );
+
+  const defaultSelectedReceiveAccountType =
+    (!values?.receiverAddress && values?.accountType === AccountTypes.Key) ||
+    (values?.receiverAddress &&
+      values?.accountType === AccountTypes.Contract &&
+      addressesEqual(providerAddress, values?.receiverAddress))
+      ? AccountTypes.Key
+      : AccountTypes.Contract;
+  
+  const [selectedReceiveAccountType, setSelectedReceiveAccountType] =
+    useState<string>(defaultSelectedReceiveAccountType);
+
+  const {
+    setTransactionBlockValues,
+    resetTransactionBlockFieldValidationError,
+    setTransactionBlockFieldValidationError,
+  } = useTransactionBuilder();
+
+  useEffect(() => {
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'offer');
+    resetTransactionBlockFieldValidationError(
+      transactionBlockId,
+      'toAssetAddress'
+    );
+  }, [selectedFromNetwork]);
+
+  const getWalletBalance = async (chainId: number, name: string) => {
+    const accountBalancePromise = [];
+    if (accountAddress && providerAddress) {
+      accountBalancePromise.push(getSupportedAssetsWithBalancesForChainId(chainId, true, accountAddress));
+      accountBalancePromise.push(getSupportedAssetsWithBalancesForChainId(chainId, true, providerAddress));
+    }
+
+    return await Promise.allSettled(accountBalancePromise)
+      .then((response: any[]) => {
+        const accountWalletBalance = response.map(accountBalance => accountBalance?.status == 'fulfilled' && accountBalance?.value);
```

Comment:
please replace whole code into `Promise.all` example in the above, would make life easier for all of us, thank you